Sprint-1/implement/sum.test.js

```diff
@@ -1,36 +1,58 @@
+const sum = require("./sum.js");
 /* Sum the numbers in an array
 
 In this kata, you will need to implement a function that sums the numerical elements of an array
 
 E.g. sum([10, 20, 30]), target output: 60
 E.g. sum(['hey', 10, 'hi', 60, 10]), target output: 80 (ignore any non-numerical elements)
 */
-
-const sum = require("./sum.js");
-
+test("only number to add sum", () => {
+  expect(sum([10, 20, 30])).toBe(60);
+  expect(sum(["hey", 10, "hi", 60, 10])).toBe(80);
+});
 // Acceptance Criteria:
 
 // Given an empty array
 // When passed to the sum function
 // Then it should return 0
-test.todo("given an empty array, returns 0")
+// test.todo("given an empty array, returns 0");
+test("empty array and return 0", () => {
+  expect(sum([])).toBe(0);
+});
 
 // Given an array with just one number
 // When passed to the sum function
 // Then it should return that number
+test("one number and return that number", () => {
+  expect(sum([3])).toBe(3);
+});
 
 // Given an array containing negative numbers
 // When passed to the sum function
 // Then it should still return the correct total sum
+test("negative number and return correct total sum", () => {
+  expect(sum([1, 2, 3, 4, -10])).toBe(0);
+});
 
 // Given an array with decimal/float numbers
 // When passed to the sum function
 // Then it should return the correct total sum
+test("add decimal float and return correct total", () => {
+  expect(sum([1.1, 2.2, 3.3, 1, 2])).toBe(9.6);
```

Comment:
Decimal numbers in most programming languages (including JS) are internally represented in "floating point number" format. Floating point arithmetic is not exact. For example, the result of `46.5678 - 46 === 0.5678` is false because `46.5678 - 46` only yield a value that is very close to `0.5678`.

So `expect(sum([1.1, 1.1, 1.1])).toEqual(3.3);` would probably fail.

Can you find a more appropriate way to test a value (that involves decimal number calculations) for "equality?

Reply:
wow!! Thanks for your comments
I use factor making 10**10 and add in my reduce second parameter with factor and return result/factor

Reply:
> The change to the code looks good, but I don't understand what you mean by `I use factor making 10**10 and add in my reduce second parameter with factor and return result/factor` 10**10 in JS means "10 to the power 10", which is equal to 10000000000.

yeah that is truth :) i change it to the whole number of decimal to avoid binary error